### ISSUE TITLE
Property Name and Value combo boxes not populated

### ISSUE THREAD

**Issue by myriammatwiy** [archimatetool/archi#1088]
### Archi Version

5.4.2

### Operating System

Windows 11

### Archi Plug-ins

coArchi 0.9.2, jArchi 1.6.1

### Description

After migrating to Archi 5.4.2 from Archi 4.9.3, the property names and values that we added and were previously displayed in the combo boxes no longer are. The names and values were transferred and are properly set on both elements and relationships. Adding new names and values in Archi 5.4.2 still works, but the new ones are also not displayed in the combo boxes. 

### Steps to reproduce

1. Select an element or a relationship
2. Select Properties under the Properties tab
3. Click the combo box beneath Name to add a property
4. Note that it only contains default names and not the ones added by users
5. Click the combo box beneath Value 
6. The Value combo box is empty
![property combo](https://github.com/user-attachments/assets/0c3cbe3b-ab48-4e78-aafe-ff76a2b2bff3)
![value combo empty](https://github.com/user-attachments/assets/8b6b070b-5bdd-4248-b2eb-30db47d934b7)

**Comment by Phillipus** [comment]
Hi, can you supply a test model where this problem occurs? You can email it to me at info (at) archimatetool.com.

I'm not sure what you mean by this:

> Note that it only contains default names and not the ones added by users

What do you mean by "default names"?

Also, are there any errors in the error log? The error log can be accessed in Archi by selecting the "About Archi" menu item. In the dialog box press the "Installation Details" button, then select the "Configuration" tab and press the "View Error Log" button.

**Comment by Franky-DG** [comment]
Hey, I have the same problem. I have a very extensive model with a lot of elements and properties. My combo box is also no longer filled, but I know the probable reason.
I think the cause can be found in 'com.archimatetool.editor/src/com/archimatetool/editor/propertysections/UserPropertiesSection.java' on line 147-148: 
```
// Maximum amount of items to display when getting all unique keys and values ​​for combo boxes
private static final int MAX_ITEMS_COMBO = 20000;
```

My model exceeded 20000 and this disrupted the functioning of the combo box.

Greetings, Franky

**Comment by Phillipus** [comment]
Hi Franky, do you have more than 20000 properties in the model? This limit is there because the combo box will be very slow to populate with this amount. You can still type in the name or value manually.

**Comment by Phillipus** [comment]
I think I know what the problem is. If you have more than 20000 elements and they all use the same property key or name they are not counted. I need to make a change to the code so it counts a max limit of unique properties.

**Comment by Phillipus** [comment]
@Franky-DG @myriammatwiy Could you test a patched version of Archi? I've uploaded a zip file containing a (hopefully) fixed version:

https://www.archimatetool.com/downloads/temp/Archi-Win64-5.4.2-test.zip

1. Download above
2. Unzip it
3. Make sure you're installed version of Archi is not running
4. Run the patched version of Archi and check the Properties combo boxes for correct entries

**Comment by Franky-DG** [comment]
Hi, I tested the patched version of Archi and it seems to have solved the problem with the contents of the combo boxes in my model.

Thanks for the quick solution.

**Comment by Phillipus** [comment]
Thanks for testing. I'll get a new version of Archi out next week.

**Comment by myriammatwiy** [comment]
> tting all unique keys and values ​​for combo box

Default names.. what I meant was that a standard list is still populated, but not the additional property names we added. Here's what we see:
![image](https://github.com/user-attachments/assets/20bac593-e05d-402b-9fc7-9011635aac4c)

We definitely don't have a list of 20000 unique properties but very likely have 20000 non-unique ones. Our org policies are not allowing me to install your patched version. I will see if I can get around that sometime today.

**Comment by myriammatwiy** [comment]
Tested and can confirm that the combos are now populated. Thanks for your prompt responses and bug fixes!

**Comment by Phillipus** [comment]
Fixed in Archi 5.4.3 released today.
